backend/src/auth/services/auth_service.py

```diff
@@ -1,82 +1,225 @@
 from typing import Optional
 
+from fastapi import HTTPException
+from fastapi.responses import JSONResponse
 from sqlalchemy.orm import Session
 
-from src.auth.exceptions import (
-    EmailTaken,
-    InvalidPassword,
-    TokenDoesNotExist,
-    UserDoesNotExist,
-    UsernameTaken,
-)
-from src.auth.models import User
-from src.auth.repositories import AuthRepository
 from src.auth.schemas import (
-    GetUser,
-    LoginResponse,
-    RefreshTokenResponse,
-    RegisterResponse,
-    UserCreate,
+    LoginEndpoint,
+    LoginUserModel,
+    RefreshTokenEndpoint,
+    RegisterEndpoint,
+    RegisterUserModel,
 )
 from src.auth.utils import (
-    decode_jwt_token,
     encode_jwt_token,
     hash_password,
     verify_passwords,
 )
-from src.core.exceptions import (
-    DetailedHTTPException,
+from src.common.exceptions import (
     NotAuthenticated,
 )
+from src.common.utils import publish_handler_event
+from src.core.interceptors.auth_interceptor import verify_user_id
+from src.users.exceptions import (
+    EmailTaken,
+    InvalidPassword,
+    InvalidUserData,
+    RegisterFailed,
+    TokenDoesNotExist,
+    UserDoesNotExist,
+    UsernameTaken,
+)
+from src.users.models.user_model import User
+from src.users.repositories import UserRepository
+from src.users.schemas.profile import CreateProfileModel
+from src.users.schemas.user import CreateUserModel
 
 
 class AuthService:
-    def __init__(self, repository: AuthRepository):
+    def __init__(
+        self,
+        repository: UserRepository,
```

Comment:
Like I said milion times, this is wrong. You should change typing to interface instead of direct class.

Reply:
I make all changes in project with importing interfaces :)